src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -308,7 +284,45 @@ private void PerformBlockPurge(IDisposable purgeRequest)
                 }
             }
 
+            if (lastMessagePurged.HasValue && OnItemPurged != null)
+            {
+                OnItemPurged(lastMessagePurged.Value);
```

Comment:
is it by design that you are calling OnItemPurged just once (outside the while loop)?